agent/src/com/cloud/agent/Agent.java

```diff
@@ -606,9 +606,8 @@ protected void processRequest(final Request request, final Link link) {
                         System.exit(1);
                         return;
                     } else if (cmd instanceof MaintainCommand) {
-                        s_logger.debug("Received maintainCommand");
-                        cancelTasks();
-                        _reconnectAllowed = false;
+                        s_logger.debug("Received maintainCommand, do not cancel current tasks");
+                        _reconnectAllowed = true;
```

Comment:
It's possible reconnectedAllowed need not be set to true or false, i.e. remove the line?

Reply:
Removed